demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -123,12 +123,21 @@ int main(int argc, char *argv[]) {
         slog::info << "Batch size is forced to  1." << slog::endl;
         cnnNetwork.setBatchSize(1);
         /** Read labels (if any)**/
-        std::string labelFileName = fileNameNoExt(FLAGS_m) + ".labels";
         std::vector<std::string> labels;
-        std::ifstream inputFile(labelFileName);
-        std::copy(std::istream_iterator<std::string>(inputFile),
-                  std::istream_iterator<std::string>(),
-                  std::back_inserter(labels));
+        std::string labelFileName;
+        if (!FLAGS_labels.empty()) {
+            labelFileName = FLAGS_labels;
+            std::ifstream inputFile(labelFileName);
+            std::string label;
+            while (std::getline(inputFile, label)) {
+                labels.push_back(label);
+            }
+        }
+        if (!labels.empty()) {
+            slog::info << "Loaded " << labels.size() << " labels" << slog::endl;
+        } else {
+            slog::info << "File " << labelFileName << " empty or not found. Labels are omitted." << slog::endl;
+        }
```

Comment:
These prints seem to be useless to me. We notice that labels are not used when the demo draws numbers instead of words. And if `-labels` is omitted intentionally it becomes even annoying because the demo reports that.

As for `labels.size()`, the size itself is not interesting and you report about labels and model output mismatch later.